pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -139,14 +139,15 @@ func (p *ConfigurationEditFlags) Apply(
 		if err != nil {
 			return errors.Wrap(err, "Invalid --env")
 		}
-		envToRemove := []string{}
-		for name := range envMap {
-			if strings.HasSuffix(name, "-") {
-				envToRemove = append(envToRemove, name[:len(name)-1])
-				delete(envMap, name)
+		envKeyToRemove := []string{}
+		for key := range envMap {
+			if !strings.HasSuffix(key, "-") {
```

Comment:
Why we did negative check than positive one above? There is no change in the behavior.

Reply:
No strong opinion is in here. It's just a matter of preference for readability, so I reverted it back.